Formula/watchman.rb

```diff
@@ -6,21 +6,13 @@ class Watchman < Formula
   head "https://github.com/facebook/watchman.git"
 
   stable do
-    url "https://github.com/facebook/watchman/archive/v4.9.0.tar.gz"
-    sha256 "1f6402dc70b1d056fffc3748f2fdcecff730d8843bb6936de395b3443ce05322"
-
-    # Upstream commit from 1 Sep 2017: "Have bin scripts use iter() method for python3"
-    patch do
-      url "https://github.com/facebook/watchman/commit/17958f7d.diff?full_index=1"
-      sha256 "edad4971fceed2aecfa2b9c3e8e22c455bfa073732a3a0c77b030e506ee860af"
-    end
+    url "https://github.com/facebook/watchman/archive/v2020.08.31.00.tar.gz"
+    sha256 "4579d17fb109a4dabb6b3ef73b44ac506e80d81b57925c6a9d80d37ed931e54e"
   end
 
-  # The Git repo contains a few tags like `2020.05.18.00`, so we have to
-  # restrict matching to versions with two to three parts (e.g., 1.2, 1.2.3).
   livecheck do
     url :head
-    regex(/^v?(\d+(?:\.\d+){,2})$/i)
+    regex(/^v?(\d+(?:\.\d+){,4})$/i)
```

Comment:
@samford @nandahkrishna suggestions here?

Reply:
If we want `livecheck` to detect the older (pre-2020 – from what I saw 2017 and earlier) tags, this seems fine to me. Sam might have other suggestions though.

Reply:
The latest `watchman` release is marked as "latest" on GitHub and we prefer to check that when it's available (and works properly). Within that context, we can simply use the standard regex for this type of check and we don't have to worry about painstakingly restricting this (like when we're working with the Git tags).

```ruby
livecheck do
  url "https://github.com/facebook/watchman/releases/latest"
  regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
end
```

Reply:
Ah I didn't spot that, thanks Sam.